src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -723,9 +724,18 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
 
       // Create the cluster
       createClusterConfig = CreateClusterConfig(machineConfig, initScript, serviceAccountInfo.clusterServiceAccount, credentialsFileName, stagingBucket, clusterScopes, clusterVPCSettings, clusterRequest.properties)
-      operation <- gdDAO.createCluster(googleProject, clusterName, createClusterConfig)
+      retryResult <- retryExponentially(whenGoogleZoneCapacityIssue, "Cluster creation failed because zone with adequate resources was not found") { () =>
+        gdDAO.createCluster(googleProject, clusterName, createClusterConfig)
+      }
+      operation <- retryResult match {
+        case Right((_, op)) => Future.successful(op)
+        case Left(errors) =>
+          Metrics.newRelic.incrementCounterIO("zoneCapacityClusterCreationFailure").unsafeRunAsync(_ => ())
```

Comment:
(my comment went away, trying again)

There are 3 possible cases here:

1. The request succeeds the first time, in which case we'll have a Right((List.empty, result))
2. The request gets retried but eventually succeeds. In which we'll have a Right((errors, result)).
3. The request gets retried and never succeeds. In which case we'll have a Left(errors).

We should probably increment the counter for cases 2 and 3, but the current code is only incrementing for case 3.